src/cascadia/TerminalApp/App.cpp

```diff
@@ -846,25 +846,27 @@ namespace winrt::TerminalApp::implementation
     // - tabViewItem: the TabViewItem in the TabView that is being removed.
     void App::_RemoveTabViewItem(const IInspectable& tabViewItem)
     {
-        // To close the window here, we need to close the hosting window.
+        uint32_t removingTabIndex = 0;
+        WINRT_ASSERT(_tabView.Items().IndexOf(tabViewItem, removingTabIndex));
+
         if (_tabs.size() == 1)
         {
+            // To close the window here, we need to close the hosting window.
             _lastTabClosedHandlers();
         }
-        uint32_t tabIndexFromControl = 0;
-        _tabView.Items().IndexOf(tabViewItem, tabIndexFromControl);
-
-        if (tabIndexFromControl == _GetFocusedTabIndex())
+        else
         {
-            _tabView.SelectedIndex((tabIndexFromControl > 0) ? tabIndexFromControl - 1 : 1);
+            // If we are closing the tab we are on, flee to an adjacent tab first!
+            // We go to the tab on the right if we are on the 0th tab, or the one on the left otherwise.
+            if (removingTabIndex == _GetFocusedTabIndex())
+            {
+                _tabView.SelectedIndex(removingTabIndex > 0 ? removingTabIndex - 1 : 1);
+            }
+            _tabView.Items().RemoveAt(removingTabIndex);

```

Comment:
Should this be moved down with line 869 like it was before? Or does closing the window (line 855) basically handle the logic of removing the XAML TabView Control?